src/v/redpanda/admin_server.cc

```diff
@@ -861,6 +861,14 @@ void admin_server::register_cluster_config_routes() {
                     pm.type = ss::sstring(p.type_name());
                 }
 
+                auto enum_values = p.enum_values();
+                if (!enum_values.empty()) {
+                    // In swagger, this field would just be called 'enum', but
+                    // because we use C++ code generation for our structures,
+                    // we cannot use that reserved word
+                    pm.enum_values = enum_values;
```

Comment:
do I understand correctly here that if pm.enum_values is given any value (including empty array) that it shows up in the output json, but otherwise isn't hidden in the json output? and since this is the same code path for all properties (not just enum properties) we don't want to add enum field to integer types etc...?

Reply:
seastar's codegen model classes omit attributes which are not set, which usually annoys me but is useful in cases like this.

Reply:
>  but otherwise isn't hidden in the json output

ahh I meant to say "is hidden".

+1